File: desktop/src/com/inkostilation/pong/desktop/display/shapes/ContainerShapes.java
package com.inkostilation.pong.desktop.display.shapes;

import com.badlogic.gdx.graphics.glutils.ShapeRenderer;

public class ContainerShapes extends AbstractShape {

    private float width, height;

    public ContainerShapes(float width, float height) {
        this.width = width;
        this.height = height;
        setReady(true);
    }

    @Override
    public DrawRect draw(DrawRect rect, ShapeRenderer renderer) {
        return rect.getNewRect(width, height);
    }
}

File: desktop/src/com/inkostilation/pong/desktop/display/shapes/ContainerShapes.kt
package com.inkostilation.pong.desktop.display.shapes

import com.badlogic.gdx.graphics.glutils.ShapeRenderer

class ContainerShapes(private val width: Float, private val height: Float) : AbstractShape() {
    override fun draw(rect: IShape.DrawRect?, renderer: ShapeRenderer): IShape.DrawRect? {
        return rect!!.getNewRect(width, height)
    }

    init {
        isReady = true
    }
}